### DISCUSSION TITLE
Import error

### DISCUSSION THREAD

**Discussion by enrico-agostini** [PMEAL/porespy#588] [Q&A] [answered]
Hi everyone, after reinstalling the newest version of `poreSpy` (v2.0.2) from anaconda (`conda install`) I tested the module on a simple python console and I the error reported below. Do you know if this is a porespy issue or something related to my Anaconda installation/python instalation?

Thans in advance,
Enrico Agostini

`In [1]: import porespy`
`---------------------------------------------------------------------------`
`IndexError                                Traceback (most recent call last)`
`<ipython-input-1-8081612b17b7> in <module>`
`----> 1 import porespy`

`~/anaconda3/lib/python3.8/site-packages/porespy/__init__.py in <module>`
`     66 from . import filters`
`     67 from . import metrics`
`---> 68 from . import networks`
`     69 from . import generators`
`     70 from . import dns`

`~/anaconda3/lib/python3.8/site-packages/porespy/networks/__init__.py in <module>`
`     25 """`
`     26 `
`---> 27 from ._funcs import add_boundary_regions`
`     28 from ._funcs import generate_voxel_image`
`     29 from ._funcs import label_phases`

`~/anaconda3/lib/python3.8/site-packages/porespy/networks/_funcs.py in <module>`
`      1 import numpy as np`
`----> 2 import openpnm as op`
`      3 import scipy.ndimage as spim`
`      4 from skimage.segmentation import find_boundaries`
`      5 from skimage.morphology import ball, cube, disk, square`

`~/anaconda3/lib/python3.8/site-packages/openpnm/__init__.py in <module>`
`     62 from . import phases`
`     63 from . import physics`
`---> 64 from . import algorithms`
`     65 from . import materials`
`     66 from . import topotools`

`~/anaconda3/lib/python3.8/site-packages/openpnm/algorithms/__init__.py in <module>`
`     11 `
`     12 from .GenericAlgorithm import GenericAlgorithm`
`---> 13 from .GenericTransport import GenericTransport`
 `    14 `
 `    15 from .ReactiveTransport import ReactiveTransport`

`~/anaconda3/lib/python3.8/site-packages/openpnm/algorithms/GenericTransport.py in <module>`
`    100 @docstr.get_sectionsf('GenericTransport', sections=['Parameters'])`
`    101 @docstr.dedent`
`--> 102 class GenericTransport(GenericAlgorithm):`
`    103     r"""`
`    104     This class implements steady-state linear transport calculations`

`~/anaconda3/lib/python3.8/site-packages/openpnm/algorithms/GenericTransport.py in GenericTransport()`
`    215         self.settings.update(**kwargs)`
`    216 `
`--> 217     @docstr.get_full_descriptionf(base='GenericTransport.reset')`
`    218     @docstr.get_sectionsf(base='GenericTransport.reset', sections=['Parameters'])`
`    219     @docstr.dedent`

`~/anaconda3/lib/python3.8/site-packages/docrep/decorators.py in deprecated(self, *args, **kwargs)`
`    125                     return getattr(self, replacement)(*args, **kwargs)`
`    126                 else:`
`--> 127                     return func(self, *args, **kwargs)`
`    128 `
`    129         else:`

`~/anaconda3/lib/python3.8/site-packages/docrep/__init__.py in get_full_descriptionf(self, *args, **kwargs)`
`    989                 removed_in="0.4.0")`
`    990     def get_full_descriptionf(self, *args, **kwargs):`
`--> 991         return self.get_full_description(base=args[0], *args[1:], **kwargs)`
`    992 `
`    993     @deprecated('get_extended_summary', "0.3.0", replace=False,`

`IndexError: tuple index out of range`

**Comment by jgostick** [comment, marked as answer]
I think you need to update your docrep package, which is used by openpnm which is imported in PoreSpy.  Do ``pip install --upgrade docrep``

**Reply by enrico-agostini** [reply]
Hi @jgostick thanks for your reply. Indeed I had to remove `openpnm` and install it again because it didn't upgrade using anaconda. Now everything seems to be working fine.
